packages/grafana-data/src/transformations/transformers/renameByRegex.ts

```diff
@@ -42,15 +42,15 @@ export const renameByRegexTransformer: DataTransformerInfo<RenameByRegexTransfor
         if (!Array.isArray(data) || data.length === 0) {
           return data;
         }
-        return data.map(renameFieldsByRegex(options));
+        return data.map(renameFieldsByRegex(options, data));
       })
     ),
 };
 
-const renameFieldsByRegex = (options: RenameByRegexTransformerOptions) => (frame: DataFrame) => {
+const renameFieldsByRegex = (options: RenameByRegexTransformerOptions, data: DataFrame[]) => (frame: DataFrame) => {
   const regex = stringToJsRegex(options.regex);
   const fields = frame.fields.map((field) => {
-    const displayName = getFieldDisplayName(field, frame);
+    const displayName = getFieldDisplayName(field, frame, data);
     if (!regex.test(displayName)) {
       return field;
     }
```

Comment:
This is sort of a bug fix for another transform that can be used with SplitBy. In the GIF example, you can see that the fields' names in the legend repeat the column name (Dog **b**, Cat, **b**), which may not be desired. You can rename fields using the 'RenameFieldsByRegex' transform, but it wasn't working how I expected.

`getFieldDisplayName` optionally takes in an array of DataFrames if the field is a part of a larger set. It uses this to distinguish fields of the same name across multiple frames. The rename fields transform wasn't passing this in, so I added it.
Now it works:
![Screen Shot 2022-08-22 at 4 17 43 PM](https://user-images.githubusercontent.com/9257800/186039664-9ee1a9f1-d92e-4fed-a28e-00100bbdc45e.png)